bonnie-businessrules/src/main/java/com/cgi/bonnie/businessrules/order/OrderStorage.java

```diff
@@ -10,15 +8,15 @@ public interface OrderStorage {
 
     boolean save(Order order);
 
-    boolean release(long id);
+ //   boolean release(long id);
 
-    boolean claim(long id, long userId);
+ //   boolean claim(long id, long userId);
 
-    boolean updateStatus(long id, Status status);
+//    boolean updateStatus(long id, Status status);
 
-    boolean setTrackingNumber(long id, String trackingNr);
+//    boolean setTrackingNumber(long id, String trackingNr);
 
-    long create(String productId, int quantity, long assignedTo, Status status);
+//    long create(String productId, int quantity, long assignedTo, Status status);
```

Comment:
I'm not a big fan of deleting these... They could be used later on. Also I'd like to have an interface with the methods, so I can assume, that someone is going to implement those functions. Maybe we should use these methods instead of modifying an order object in the business rules and save it at the end.

Reply:
The business rules has to control/check the content  of a new order, before creating/storing it.
Whit this method possible to create/store an AssemblyOrder without content check and without shopOrderId and/or plecamentDate and/or metadata information. But it won't converted to Order     whithout necessary fields ... 
I prefer creating/storing data only whit consistent contents (with a check() method before save/create).
i.e. checking rules
    without shopOrderId can't notice the shop by changing its state 
    without metadata can't delivery
    without assignedTo can't modify/store the state to 'claimed'
    ...

Reply:
Yes, that's right, for this method we should rather use an Order object, so this particular method can be deleted (and also all the corresponding ones, so we can use the createOrder(Order order) method everywhere).
Although, I wouldn't delete the other methods you commented, because of the reasons I mentioned earlier